### ISSUE TITLE
Allow comments in JSON .stylelintrc file

### ISSUE THREAD

**Issue by alawatthe** [stylelint/stylelint#982]
I'd like to use comments in the json of my  ```.stylelintrc```file. (Yes, I know that this is not valid json, but it is useful, e.g. to describe rules a bit, explain why they were added, temporarily comment out rules, ...).

Right now, the following ```.stylelintrc```file:
```
{
  "rules": {
    "color-hex-case": "upper",
    "color-hex-length": "long",
    "color-named": "never",
    // "color-no-hex": false,
    "color-no-invalid-hex": true,
  }
}
```
gives the error
```
Error: Undefined rule "// "color-no-hex""
```

> Which version of stylelint are you using?
5.2.1

**Comment by TrySound** [comment]
@alawatthe If you want to add comments, use language where comments are valid. Comments in json is hack.
https://nodeca.github.io/js-yaml/

**Comment by alawatthe** [comment]
@TrySound I'd like to disagree on that one. Perhaps comments in json are a bit hacky (I don't think they are), but they are very useful (see my message above).

All other linters I know support comments in json (see [here](http://eslint.org/docs/user-guide/configuring#configuration-file-formats) for example). And I don't think there is a reason to not support this, since implementing this wouldn't be a big problem. Using [strip-json-comments](https://github.com/sindresorhus/strip-json-comments) it is just a

```
JSON.parse(stripJsonComments(json));
```

instead of 

```
JSON.parse(json);
```

I can send a pull request if you want.

Forcing the user of stylelint to use a entirely different language for the configuration files, because of a problem, which can be solved easily is not the best thing to do.

On more thing on the comments in JSON are a hack: to [quote Douglas Crockford](https://plus.google.com/+DouglasCrockfordEsq/posts/RK8qyGVaGSr) (the inventor of JSON):

> Suppose you are using JSON to keep configuration files, which you would like to annotate. Go ahead and insert all the comments you like. Then pipe it through JSMin before handing it to your JSON parser.﻿

Which is exactly what I want to do.

**Comment by TrySound** [comment]
Go ahead. We will discuss there.

**Comment by alawatthe** [comment]
Moving discussion to https://github.com/davidtheclark/cosmiconfig/issues/20

**Comment by davidtheclark** [comment]
To be clear for people coming across this issue: **you *can* use comments in stylelint configuration files**. Because your configuration file does not have to be JSON.

YAML and JS syntaxes allow comments. JSON syntax does not. And right now JSON is strictly parsed, so comments are not allowed.

Discussion about parsing comments in JSON will be over in the cosmiconfig issue.

**Comment by davidtheclark** [comment]
Sorry, will leave open for a bit for other input.

**Comment by alawatthe** [comment]
I am now using the js way of writing the config files (this was somehow not working earlier, that's why I used the JSON way in the first place), I don't need the comments in JSON functionality anymore.

Sorry for the noise in the issue tracker :-/ and thank you for the nice tool :-)

- - - 
As an aside: for everyone wanting to use comments with JSON, it can be done like this:
```js
const parseJson = require('parse-json');
const stripJsonComments = require('strip-json-comments');
const stylelintrc = parseJson(stripJsonComments(fs.readFileSync('.stylelintrc', 'utf8')));

// ...
stylelint(stylelintrc)
// ...
```

**Comment by davidtheclark** [comment]
@alawatthe: Cool. I'm sure you're not the first or last person with the same suggestion, so it will continue to be a possibility.
